pkg/sensors/tracing/generickprobe.go

```diff
@@ -284,6 +288,25 @@ func createGenericKprobeSensor(name string, kprobes []v1alpha1.KProbeSpec) (*sen
 
 		argRetprobe = nil // holds pointer to arg for return handler
 		funcName := f.Call
+		// if kprobe call is a syscall, we try to detect if the user
+		// specified a prefix in the form of "__x64" or "__arm64", and
+		// if not, transparently add it based on runtime.GOARCH
+		if f.Syscall {
+			alreadyPrefixed := false
+			for _, p := range supportedArchPrefix {
+				if strings.HasPrefix(f.Call, p) {
+					alreadyPrefixed = true
+					break
+				}
+			}
+			if !alreadyPrefixed {
+				if prefix, found := supportedArchPrefix[runtime.GOARCH]; found {
+					funcName = prefix + "_" + funcName
+				} else {
+					logger.GetLogger().Warnf("kprobe syscall prefix detection, unsupported architecture: %s", runtime.GOARCH)
+				}
+			}
```

Comment:
Can we add a warning here if the existing prefix does not match the arch?
Would be nice to move that code into a function and write a couple of simple unit tests. 

I think we also need to change btf validation code: https://github.com/cilium/tetragon/blob/main/pkg/btf/validation.go#L128.

Reply:
- I moved this thing into a function and now it's way better
- For btf validation code, I change `f.Call` var directly so that prefix appending is transparent to this code.